### PR TITLE
[#266] Set `minimum-stability` to `stable` and added docs.

### LINKED ISSUE #266: Set minimum-stability to "stable"
Currently minimum-stability is set to "dev".
I think we should rely on composer defaults and have the minimum stability set to "stable" just by removing that line.
https://getcomposer.org/doc/04-schema.md#minimum-stability

Then we will need to add dev, alpha, beta, RC suffixes to the specific dependencies that need it like for example devel or drupal console. 
https://getcomposer.org/doc/04-schema.md#package-links
The composer.json get also more semantic.

This way we don't hide dependencies on modules, libraries, etc that are not so mature.

This all started for me because there is a contrib module with a broken HEAD
https://www.drupal.org/node/2872222

What do you think?

### PR DESCRIPTION
**Pull request by AlexSkrypnyk**
Closes #266

### CODE REVIEW

File: composer.json
Comment (AlexSkrypnyk) [inline composer.json]: This package was updated to Symfony 7, but there is no release.

Considering that the whole `11.x` branch is "early preview", I think this should be okay to have like this.

Once Drupal 11 is in `beta` and all these packages have their releases, we can clean it up.

File: composer.json
Comment (AlexSkrypnyk) [inline composer.json]: This package was updated to Symfony 7, but there is no release.

Considering that the whole 11.x branch is "early preview", I think this should be okay to have like this.

Once Drupal 11 is in beta and all these packages have their releases, we can clean it up.

I've created an [issue](https://github.com/Chi-teck/drupal-code-generator/issues/152) - maybe we can wait for a couple of days before merging this one.

Comment (simesy) [reply composer.json]: I don't think you need to add `as 3.5.0` unless another package explicitly needs that version. If you are using `composer require ...` it won't work because it checks stability. But if you just add it manually to the composer.json it works fine. This is a feature afaik.

To test, I updated it locally to be this and ran composer update. 

```
 "chi-teck/drupal-code-generator": "4.x-dev",
```

I don't know if pinning it to a commit hash is good... if the package maintainers happen to rewrite history on the dev branch then composer will stop building?

File: composer.json
Comment (simesy) [inline composer.json:39]: I like this change.

File: composer.json
Comment (AlexSkrypnyk) [reply composer.json]: it worked for you because you probably did not use the `dev` version of core.

Drush 13.x requires `4.x-dev` for Symfony 7 compatibility https://github.com/drush-ops/drush/blob/13.x/composer.json#L36C3-L36C57

Comment (AlexSkrypnyk) [reply composer.json]: `chi-teck/drupal-code-generator` is now released. waiting for Drush to start using it at `alpha+` version level.

**Review by normanlolx — APPROVED** [review]
Nice!! Very clear documentation, I like. I'm also in favor of waiting for new code generator release instead of getting the alias committed. But when there's a follow-up issue so we don't forget about it I'm also okay with it in the meanwhile.

File: composer.json
Comment (AlexSkrypnyk) [reply composer.json]: https://github.com/drush-ops/drush/pull/6010

**Comment by AlexSkrypnyk** [comment]
@leymannx 
We now have an issue with Drush (https://github.com/drush-ops/drush/pull/6010). I'm going to wait for a couple of days before merging the alias (ugly).

File: composer.json
Comment (simesy) [reply composer.json]: > it worked for you because you probably did not use the dev version of core.

Sorry I isn't sinking in why this project needs to use the dev version of core.

**Pull request merged**
